lib/internal/errors.js

```diff
@@ -1122,7 +1129,26 @@ E('ERR_OUT_OF_RANGE',
     msg += ` It must be ${range}. Received ${received}`;
     return msg;
   }, RangeError);
-E('ERR_REQUIRE_ESM', 'Must use import to load ES Module: %s', Error);
+E('ERR_REQUIRE_ESM', (filename, module = null, pkg = null) => {
```

Comment:
Probably best to move the logic out and just make this a function of `(filename, parentPath, packageJsonPath)`.

Reply:
Sorry. I'm not understanding

Reply:
So calling the error function with:

```js
const parentPath = module.parent && module.parent.filename;
const packageJsonPath = path.resolve(pkg.path, 'package.json');
ERR_REQUIRE_ESM(filename, parentPath, packageJsonPath);
```

This ensures only string logic is needed in the error function, and encapsulates the error function from needing to be dependent on the more complex object structures as well.